### ISSUE TITLE
place text on line 

### ISSUE THREAD

**Issue by Johnlon** [wavedrom/wavedrom#302]
Online editor.

I want to put text along a line in a similar way to how "data" works.
However, I want the text on high or low lines - I don't want the text shown inside a data lozenge.

I can approximate what I want using edges but the problem here is that I don't want to see the lines of the edges.
![wd](https://user-images.githubusercontent.com/836248/87865662-01169a00-c970-11ea-8901-4f1473625943.png)

**Comment by t-moe** [comment]
Yes. Please add labels for low/high lines (or any state?) similar to what "data" provides for "values".

I think this is really important and it was the first thing I was looking for.
Without it, I can't even label a simple RS232 transmission start bit.

**Comment by drom** [comment]
Is the example below close to what you trying to achieve?
Empty extra lane explicitly introduced for labels:

```wavedrom
{signal: [
  {name: 'foo', wave: 'p....'},
  {                            node: '.A..B.'},
  {name: 'bar', wave: '01..0', node: '.a..b.'},
],
  edge: ['a-A', 'A-B some text', 'B-b']
}
```

![image](https://user-images.githubusercontent.com/511872/90346354-f423af80-dfdc-11ea-8198-7f9aa6fdb88b.png)

**Comment by t-moe** [comment]
I would suggest a simpler approach:
![labelsExample](https://user-images.githubusercontent.com/11081135/90370567-e5d1a400-e06d-11ea-8008-ab312b4aede3.png)

Maybe, you could add a 'alldata' property, to not break the functionality of the existing data property.
```
{signal: [ 
    {name: 'UART Example', 'wave': '104.......1u', 'alldata': ['','Start','data', 'ACK', '']},
]}

```

**Comment by drom** [comment]
I understand now. Allowing `data` attribute on 0,1 and other segments do make sense.
It should not break anything. Will fix.

**Comment by t-moe** [comment]
I'm having a hard time navigating through the source code, as large parts are undocumented. I couldn't pinpoint the location where you decide whether or not to render a label at a certain point in time.
I would really appreciate it if you could provide a fix.

**Comment by Johnlon** [comment]
> I would suggest a simpler approach:
> ![labelsExample](https://user-images.githubusercontent.com/11081135/90370567-e5d1a400-e06d-11ea-8008-ab312b4aede3.png)
> 
> Maybe, you could add a 'alldata' property, to not break the functionality of the existing data property.
> 
> ```
> {signal: [ 
>     {name: 'UART Example', 'wave': '104.......1u', 'alldata': ['','Start','data', 'ACK', '']},
> ]}
> ```

Yes that aligns with my need in my original post

**Comment by ColePBryan** [comment]
@drom I have the code to match the above suggested implementation from @Johnlon . The one thing I am wondering is if this should be implemented in a backwards compatible way? It seems like this could break a lot of existing wavefiles when this is integrated. We could possibly make so it can be enabled in the config?

**Comment by ColePBryan** [comment]
Actually I just saw the @Johnlon changed it from 'data' to 'alldata' which would be backwards compatible.

**Comment by ColePBryan** [comment]
With the all data approach how should it be handled when both 'data' and 'alldata' are provided?

**Comment by ColePBryan** [comment]
Here is an initial PR that would add the following functionality:

```json5
{
 signal: [ 
    {name: 'UART Example', 'wave': '104.......1u', 'data': ['','Start','data', 'ACK', '']}
    ],
 config: {'labelsOnOneAndZeroLines':true}
}
```

![image](https://user-images.githubusercontent.com/20587384/102177066-95b9f100-3e57-11eb-9c84-748c87fa80da.png)
